sw/device/tests/pattgen_ios_test.c

```diff
@@ -0,0 +1,276 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_pattgen.h"
+#include "sw/device/lib/dif/dif_pinmux.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/irq.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_console.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/lib/testing/test_framework/ottf_utils.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "pattgen_regs.h"  // Generated.
+
+/**
+   PATTGEN IOS test
+   The test programs both channels of the pattgen to random patterns.
+   Pattgen ios are connected as follows:
+   pda_tx[0] - IOB9
+   pcl_tx[0] - IOB10
+   pda_tx[1] - IOB11
+   pcl_tx[1] - IOB12
+
+   The random pattern programmed by 'chip_sw_patt_ios_vseq.sv'
+   and output will be collected at the PADS and compared in SV test bench.
+ */
+
+OTTF_DEFINE_TEST_CONFIG(.enable_uart_flow_control = true);
+
+enum {
+  kTestCmdTimeoutUsec = 5000000,
+};
+
+/* Macro to define and use variable that need to be stored in flash
+ * for DV tests and RAM for real tests, to enable backdoor access. */
+#define DEFINE_BACKDOOR_VAR(type, name, default_val)                       \
```

Comment:
This seems like something that would be useful in `sw/device/lib/base/macros.h`, thoughts?

Reply:
(OK to do in follow up PR)

Reply:
Yes that's a good idea, we probably want to revisit how the backdoor variables are handled, but it's better done in another PR to convert all tests.